addons/sale/controllers/portal.py

```diff
@@ -244,7 +244,7 @@ def portal_quote_accept(self, order_id, access_token=None, name=None, signature=
 
         if not order_sudo._has_to_be_paid():
             order_sudo.action_confirm()
-            order_sudo._send_order_confirmation_mail()
+            order_sudo._send_payment_succeeded_for_order_mail()
```

Comment:
There is no payment logic here, we only confirmed a SO by signing it.
We get the impression that you are sending a 'payment succeeded' mail when no payment happened at all.

Okay, the logic behind it will send the right mail, but it's confusing to read and understand.